core/Scheduler/Scheduler.php

```diff
@@ -118,6 +125,14 @@ public function run()
                     continue;
                 }
 
+                $taskName = $task->getName();
+
+                if (!$this->acquireLockForTask($taskName, $task->getTTL())) {
+                    $this->logger->debug("Scheduler: '{task}' is currently executed by another process",
+                        ['task' => $task->getName()]);
+                    continue;
+                }
+
                 if ($readFromOption) {
```

Comment:
Is this block needed with the new lock mechanism? It sounds confusing and the long comment is not helping either.

Reply:
I'm actually not 100% sure it's still needed. But if two processes run the scheduled tasks in parallel it might be better to read the current state from the option again, as it might already have been updated by the other process.
If for example the option has stored that Task X should run now, a first process might execute it and update the option, that Task X shouldn't run now anymore. A second process, that might still be busy with another task, should update the plan from the option again, to ensure the plan wasn't updated meanwhile, as otherwise Task X might run again.
Hope that makes sense.

Reply:
Yeah, makes some sense as a secondary mechanism/a safeguard after the lock.